pkg/monitors/http/http.go

```diff
@@ -163,18 +167,21 @@ func (m *Monitor) normalizeURL(site string) (normalizedURL *url.URL, err error)
 	if err != nil {
 		return
 	}
+	host := stringURL.Hostname()
 	port := stringURL.Port()
-	if port == "" {
+	// for deprecated URLs only, set default port if not explicitly set
+	if host == stringURL.Host {
```

Comment:
What is the difference between `Host` and `Hostname()`?

Also couldn't the port be 443 if the scheme is https?

Reply:
> What is the difference between Host and Hostname()?

for https://signalfx.com:443
Host = `signalfx.com`
Hostname() = `signalfx.com:443`

> Also couldn't the port be 443 if the scheme is https?

The goal is to avoid to set the port when it is default.
It will fix the error we do not have with `curl` or `python/requests`.

But we can add the default port 443 explicitly here and remove the port before to do the request. it will also work but default port will remain in `url` dimension.

Reply:
sorry it is the opposite:

Hostname() = `signalfx.com`
Host = `signalfx.com:443`

Reply:
in short, we need to not have default port in URL when we do the request from `net/http` for a consistent behavior with curl & co.

Here I choose to remove default port from general url (used for everywhere: dimension and request). we can do it only for request but this better the have same value on dimension too.